pkg/agent/protocol/http/proxy_test.go

```diff
@@ -371,35 +313,54 @@ func Test_ProxyHandler(t *testing.T) {
 		t.Run(tc.title, func(t *testing.T) {
 			t.Parallel()
 
-			client := &fakeHTTPClient{
-				body:    tc.body,
-				status:  tc.expectedStatus,
-				headers: tc.headers,
-			}
+			upstreamServer := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+				for k, values := range tc.headers {
+					for _, v := range values {
+						rw.Header().Add(k, v)
+					}
+				}
+				rw.WriteHeader(tc.statusCode)
 
-			upstreamURL, err := url.Parse(tc.config.UpstreamAddress)
+				_, err := rw.Write(tc.body)
+				if err != nil {
+					t.Errorf("writing upstream body: %v", err)
+				}
+			}))
+
+			upstreamURL, err := url.Parse(upstreamServer.URL)
 			if err != nil {
-				t.Errorf("error parsing upstream address %v", err)
-				return
+				t.Fatalf("error parsing httptest url")
 			}
+
 			handler := &httpHandler{
 				upstreamURL: *upstreamURL,
-				client:      client,
+				client:      http.DefaultClient,
 				disruption:  tc.disruption,
 				metrics:     &protocol.MetricMap{},
 			}
 
-			reqURL := fmt.Sprintf("http://%s%s", tc.config.ListenAddress, tc.path)
-			req := httptest.NewRequest(tc.method, reqURL, strings.NewReader(string(tc.body)))
-			recorder := httptest.NewRecorder()
-			handler.ServeHTTP(recorder, req)
-			resp := recorder.Result()
+			proxyServer := httptest.NewServer(handler)
+
+			req, err := http.NewRequest(tc.method, proxyServer.URL+tc.path, bytes.NewReader(tc.body))
```

Comment:
Reading this in the diff, it is not fully clear to me whether `tc.body` was intended to be the request body in the first place, specially given that all requests are (currently) `GET` requests. Should I get rid of this and add a new `requestBody` field to the test table?

Reply:
The `tc.body` field was intended to be used in requests that were not `GET`, which as you point out, are still missing in the test cases, but we should consider adding. 
 
What I don't understand is your proposal of getting rid of `tc.body` and adding a new `requestBody` field to the test table. Wouldn't it be just a rename of `body` by `requestBody`?

Reply:
Maybe we're talking about different things, in the diff I see:

> `req := httptest.NewRequest(tc.method, reqURL, strings.NewReader(string(tc.body)))`

Which is the test case body being used as a request body, that indeed sounds like a bug in the test.

Reply:
If I understand the code correctly, the request we're performing here would be the client request, meaning the first request of the flow. Am I right? If this is the case, I believe the body for that request belongs to the test case (if we want to add it).

Reply:
I've renamed the fields to clarify these are used to control the upstream server response, and fixed the bug where they were being sent by the client https://github.com/grafana/xk6-disruptor/pull/308/commits/371bb82319354c8bcb97b0ef34cdc27d7cfded92 

Does this align with what you said?